receiver.go

```diff
@@ -344,9 +344,14 @@ func (r *Receiver) listenForMessage(ctx context.Context) (*amqp.Message, error)
 	ctx, span := r.startConsumerSpanFromContext(ctx, "sb.Receiver.listenForMessage")
 	defer span.End()
 
+	var receiver *amqp.Receiver
 	r.clientMu.RLock()
-	msg, err := r.receiver.Receive(ctx)
+	if r.receiver == nil {
+		return nil, r.connClosedError(ctx)
+	}
+	receiver = r.receiver
 	r.clientMu.RUnlock()
+	msg, err := receiver.Receive(ctx)
```

Comment:
Receive is out of the lock now, is that right?

Reply:
Correct